src/librustdoc/passes/doc_test_lints.rs

```diff
@@ -58,23 +60,30 @@ impl crate::doctest::Tester for Tests {
 }
 
 crate fn should_have_doc_example(cx: &DocContext<'_>, item: &clean::Item) -> bool {
-    if matches!(
-        *item.kind,
-        clean::StructFieldItem(_)
-            | clean::VariantItem(_)
-            | clean::AssocConstItem(_, _)
-            | clean::AssocTypeItem(_, _)
-            | clean::TypedefItem(_, _)
-            | clean::StaticItem(_)
-            | clean::ConstantItem(_)
-            | clean::ExternCrateItem(_, _)
-            | clean::ImportItem(_)
-            | clean::PrimitiveItem(_)
-            | clean::KeywordItem(_)
-    ) {
+    if !cx.renderinfo.borrow().access_levels.is_public(item.def_id)
+        || matches!(
+            *item.kind,
+            clean::StructFieldItem(_)
+                | clean::VariantItem(_)
+                | clean::AssocConstItem(_, _)
+                | clean::AssocTypeItem(_, _)
+                | clean::TypedefItem(_, _)
+                | clean::StaticItem(_)
+                | clean::ConstantItem(_)
+                | clean::ExternCrateItem(_, _)
+                | clean::ImportItem(_)
+                | clean::PrimitiveItem(_)
+                | clean::KeywordItem(_)
+        )
+    {
         return false;
     }
     let hir_id = cx.tcx.hir().local_def_id_to_hir_id(item.def_id.expect_local());
+    if cx.tcx.hir().attrs(hir_id).lists(sym::doc).has_word(sym::hidden)
+        || inherits_doc_hidden(cx, hir_id)
+    {
+        return false;
+    }
```

Comment:
Shouldn't this be handled by `access_levels.is_public()`? What happens when you remove this check?

Reply:
```
error: missing code example in this documentation
  --> /home/imperio/rust/rust/src/test/rustdoc-ui/lint-missing-doc-code-example.rs:19:1
   |
LL | / pub mod module1 { //~ ERROR
LL | | }
   | |_^
   |
note: the lint level is defined here
  --> /home/imperio/rust/rust/src/test/rustdoc-ui/lint-missing-doc-code-example.rs:2:9
   |
LL | #![deny(missing_doc_code_examples)]
   |         ^^^^^^^^^^^^^^^^^^^^^^^^^

error: missing code example in this documentation
  --> /home/imperio/rust/rust/src/test/rustdoc-ui/lint-missing-doc-code-example.rs:37:3
   |
LL |   /// doc
   |   ^^^^^^^

error: missing code example in this documentation
  --> /home/imperio/rust/rust/src/test/rustdoc-ui/lint-missing-doc-code-example.rs:75:1
   |
LL | / pub mod foo {
LL | |     pub fn bar() {}
LL | | }
   | |_^

error: missing code example in this documentation
  --> /home/imperio/rust/rust/src/test/rustdoc-ui/lint-missing-doc-code-example.rs:76:5
   |
LL |     pub fn bar() {}
   |     ^^^^^^^^^^^^^^^

error: missing code example in this documentation
  --> /home/imperio/rust/rust/src/test/rustdoc-ui/lint-missing-doc-code-example.rs:49:1
   |
LL | /// Doc
   | ^^^^^^^

error: missing code example in this documentation
  --> /home/imperio/rust/rust/src/test/rustdoc-ui/lint-missing-doc-code-example.rs:56:1
   |
LL | /// Doc
   | ^^^^^^^

error: missing code example in this documentation
  --> /home/imperio/rust/rust/src/test/rustdoc-ui/lint-missing-doc-code-example.rs:63:1
   |
LL | /// Doc
   | ^^^^^^^

error: aborting due to 7 previous errors
```